fluent/fluent.go

```diff
@@ -297,6 +301,12 @@ func (f *Fluent) send() error {
 
 	var err error
 	if len(f.pending) > 0 {
+		t := f.Config.WriteTimeout
+		if time.Duration(0) < t {
+			f.conn.SetWriteDeadline(time.Now().Add(t))
+		} else {
+			f.conn.SetWriteDeadline(time.Time{})
```

Comment:
I have no idea about what this line does, but, in my feeling, having no "else" clause looks the same behavior with previous. Could you explain what this line does?